api/Services/LocationService.cs

```diff
@@ -50,6 +54,24 @@ IAppCache cache
 
         #region Collection Methods
 
+        public async Task<ICollection<Location>> GetPCSSLocationsAndCourtRooms() => await GetDataFromCache("PCSSLocationsAndCourtRooms", async () =>
+        {
+            var locations = await _pcssLookupServiceClient.GetCourtRoomsAsync();
+
+            var mappedLocations = locations
+                .Select(l => new Location
+                {
+                    LocationId = l.LocationId.ToString(),
+                    Name = l.LocationNm,
+                    Code = l.LocationSNm,
+                    CourtRooms = l.CourtRooms?.Select(cr => new CourtRoom { Room = cr.CourtRoomCd }).ToList() ?? new List<CourtRoom>()
+                })
+                .OrderBy(l => l.Name)
+                .ToList();
+
+            return mappedLocations;
+        });
```

Comment:
I'd recommend using a mapping framework like AutoMapper for this.  You can find configuration examples in the DARS source code `DARS.Server.Data.Mapper`